app/src/ui/app.tsx

```diff
@@ -1781,6 +1781,7 @@ export class App extends React.Component<IAppProps, IAppState> {
         selectedRepository={selectedRepository}
         onSelectionChanged={this.onSelectionChanged}
         repositories={this.state.repositories}
+        recentRepositories={this.props.appStore.getRecentRepositories()}
```

Comment:
Would it be possible to have this list stored in `this.state`? I worry that using functions on props like this to manage data is an anti-pattern, when we have an existing way to manage this.

Reply:
An earlier PR where I discussed this pattern (which we currently do in a couple of places): https://github.com/desktop/desktop/pull/6929

Reply:
ooo good point! will update!

Reply:
what do you think about b3002e8?

Reply:
Much nicer! 🎨